src/system/Form/Input.js

```diff
@@ -36,21 +46,22 @@ const Input = React.forwardRef(
 					{ label }
 				</Label>
 			) }
-
-			<ThemeInput
-				ref={ ref }
-				id={ forLabel }
-				required={ required }
-				aria-required={ required }
-				aria-describedby={ hasError ? `describe-${ forLabel }-validation` : undefined }
-				sx={ {
-					...inputStyles,
-					...sx,
-					...( hasError ? { borderColor: 'input.border.error' } : {} ),
-				} }
-				{ ...props }
-			/>
-
+			<div sx={ flexComponent && flexComponent.display === 'flex' ? { display: 'flex' } : {} }>
```

Comment:
@kat3samsin this works 🎉 . I was thinking if we need the extra extensibility here, since I see this being used mostly for input + copy. Should we have a dedicate component that encapsulate the input, instead of adding options to this one?

The main benefit I see in case we don't need the extra stuff and can stay with a "simple" component that is a "InputWithCopyButton", is that the declaration and the code we'll write would be probably easier to understand and we would not need to pass extra components to the Design System, since they would all belong there.

Reply:
This actually makes a lot of sense! I'll update to create a new component, thanks for the suggestion!